main/src/ca/uwaterloo/flix/language/phase/Weeder.scala

```diff
@@ -116,7 +116,11 @@ object Weeder {
 
     case d: ParsedAst.Declaration.Instance => visitInstance(d)
 
+    case _: ParsedAst.Declaration.Effect => Nil.toSuccess // ignoring effect declarations for now
+
     case _: ParsedAst.Declaration.Sig => throw InternalCompilerException(s"Unexpected declaration")
+
+    case _: ParsedAst.Declaration.Op => throw InternalCompilerException(s"Unexpected declaration")
```

Comment:
Does this crash the compiler or can it not happen?

Reply:
Can't happen. This visitor is at the top level, where the parser will never allow an operation.

Maybe we can remove Sig and Op from the Declaration enum.